hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferContainer.java

```diff
@@ -464,8 +520,11 @@ public void writeData(ObjectDataOutput out) throws IOException {
     @Override
     @SuppressWarnings("unchecked")
     public void readData(ObjectDataInput in) throws IOException {
-        name = in.readUTF();
-        emptyRingWaitNotifyKey = new RingbufferWaitNotifyKey(name, EMPTY);
+        assert !in.getVersion().isUnknown();
```

Comment:
Aha, so the version retrieved in `ObjectDataInput#getVersion` is set from the version in the serialised `byte[]`, not the `Cluster#getClusterVersion`? I thought it was the latter.